From 84b2ddf1b8aa4803fc4ab14420b578f2dc7df2ac Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 10 Nov 2022 14:39:54 +0200
Subject: [PATCH] [SW-105079] gaudi3: pldm: configure network pll

In order to speed up ppboot execution, preboot loading time
and execution time on sram this patch performs router
and mesh pll configuration (MESH, NCH, C2C) only for pldm.

Change-Id: Ie2a167dd3f7afc74907d0ed322d839083aa15c36
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index c214089..5585f43 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -169,6 +169,12 @@
 	init_prstn();
 	hl_dev_cpu_pll_init();
 }
+__weak void hl_dev_nw_pll_init(void) {}
+
+void hl_plat_init(void)
+{
+	hl_dev_nw_pll_init();
+}
 
 void preboot_sync_btl(void)
 {
diff --git a/arch/gaudi3/arc/pldm.c b/arch/gaudi3/arc/pldm.c
index 0e4ea91..ba185d2 100644
--- a/arch/gaudi3/arc/pldm.c
+++ b/arch/gaudi3/arc/pldm.c
@@ -34,6 +34,59 @@
 	}
 };
 
+static struct hl_pll_cfg gaudi3_nw_pll_cfg[] = {
+	{
+		.name = "G1C2C", .base = mmD0_G1_C2C_PLL_CTRL_BASE,
+		.nr = 1, .nf = 224, .od = 6, .nb = 0,
+		.div_factor = {0, 1, 1, 7},
+		.div_sel = {HL_PLL_PLL_CLK,
+			HL_PLL_DIV_PLL,
+			HL_PLL_DIV_PLL,
+			HL_PLL_DIV_PLL
+		},
+	},
+	{
+		.name = "G5C2C", .base = mmD0_G5_C2C_PLL_CTRL_BASE,
+		.nr = 1, .nf = 224, .od = 6, .nb = 0,
+		.div_factor = {0, 1, 1, 7},
+		.div_sel = {HL_PLL_PLL_CLK,
+			HL_PLL_DIV_PLL,
+			HL_PLL_DIV_PLL,
+			HL_PLL_DIV_PLL
+		},
+	},
+	{
+		.name = "G4NCH", .base = mmD0_G4_NCH_PLL_CTRL_BASE,
+		.nr = 1, .nf = 224, .od = 6, .nb = 0,
+		.div_factor = {0, 1, 1, 7},
+		.div_sel = {HL_PLL_PLL_CLK,
+			HL_PLL_DIV_PLL,
+			HL_PLL_DIV_PLL,
+			HL_PLL_DIV_PLL
+		},
+	},
+	{
+		.name = "G0MSH", .base = mmD0_G0_MSH_PLL_CTRL_BASE,
+		.nr = 1, .nf = 224, .od = 6, .nb = 0,
+		.div_factor = {0, 1, 1, 7},
+		.div_sel = {HL_PLL_PLL_CLK,
+			HL_PLL_DIV_PLL,
+			HL_PLL_DIV_PLL,
+			HL_PLL_DIV_PLL
+		},
+	},
+	{
+		.name = "G6MSH", .base = mmD0_G6_MSH_PLL_CTRL_BASE,
+		.nr = 1, .nf = 224, .od = 6, .nb = 0,
+		.div_factor = {0, 1, 1, 7},
+		.div_sel = {HL_PLL_PLL_CLK,
+			HL_PLL_DIV_PLL,
+			HL_PLL_DIV_PLL,
+			HL_PLL_DIV_PLL
+		},
+	},
+};
+
 u64 hl_pll_timestamp_freq_get(void)
 {
 	return SYS_CLOCK_FREQUENCY;
@@ -117,3 +170,7 @@
 	hl_pll_init(gaudi3_cpu_pll_cfg, 1);
 }
 
+void hl_dev_nw_pll_init(void)
+{
+	hl_pll_init(gaudi3_nw_pll_cfg, ARRAY_SIZE(gaudi3_nw_pll_cfg));
+}
